src/components/organisms/PaymentModal/PaymentModal.tsx

```diff
@@ -55,23 +58,21 @@ const PaymentModal: React.FunctionComponent<PropsType> = ({
   let modalContent;
   if (!purchaseHistoryRequestStatus) {
     modalContent = <>Loading...</>;
-  } else if (isPaymentSuccess) {
+  } else if (eventPaidSuccessfully === selectedEvent.id) {
     modalContent = (
       <PaymentConfirmation startUtcSeconds={selectedEvent.start_utc_seconds} />
     );
   } else if (!hasUserBoughtTicket) {
     modalContent = (
       <PaymentForm
-        setIsPaymentSuccess={setIsPaymentSuccess}
+        setIsPaymentSuccess={() => setEventPaidSuccessfully(selectedEvent.id)}
         setIsPaymentProceeding={setIsPaymentProceeding}
         setIsCardBeingSaved={setIsCardBeingSaved}
         isPaymentProceeding={isPaymentProceeding}
         isCardBeingSaved={isCardBeingSaved}
         event={selectedEvent}
       />
     );
-  } else {
```

Comment:
Why remove this? It shouldn't be possible but who knows.

Reply:
Or always display PaymentConfirmation if `hasUserBoughtTicket`. Here it seems we have a case where the users would not see anything